content/algorithms/Dijkstra/Dijkstra.md

```diff
@@ -0,0 +1,241 @@
+            elif u not in seen or vu_dist < seen[u]:
+                seen[u] = vu_dist
+                push(fringe, (vu_dist, next(c), u))
+                if paths is not None:
+                    paths[u] = paths[v] + [u]
+
+# printing the distance and path from the source node 'DEL' to target node 'LCY'                    
+print(dist[target], paths[target])
+```
+
+Don't worry, you don't need to write all this code again and again. NetworkX got you covered!! So, NetworkX provides a lot of functions with the help of which one can actually find the [shortest path](https://networkx.org/documentation/stable/reference/algorithms/shortest_paths.html) based on their needs. 
+
+All functions using dijkstra's algorithm are similar, but, for this example the most suitable is [single_source_dijkstra()](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.shortest_paths.weighted.single_source_dijkstra.html#networkx.algorithms.shortest_paths.weighted.single_source_dijkstra). It comes out that this function actually gives the same output as the one calculated in the above example.
+
+```python
+nx.single_source_dijkstra(flight_path, 'DEL', target='LCY', weight="weight")
+```
+
+## Applications of Dijkstra's Algorithm
+
+It is used as a part of applications to find the shortest path if required. There are other factors as well which are considered in every application while implementing Dijkstra's algorithm. Like,
+- In special drones or robots for delivery service, it is used as a part to identify the shortest path possible.
+- One of the most common use case is Google Maps. It helps to find the best route possible in shortest time.
+- In social media applications, for smaller graphs it can be used effectively to suggest the "people you may know" section.
+- As the above example, it can be used in a software which calculates and informs the estimate arrival time, best route etc. of a flight to a user.
+- It is used in IP routing to find Open shortest Path First.
+- It is used in the telephone network.
+
+
+## Advantages and Disadvantages of Dijkstra's Algorithm
+
+<b>ADVANTAGES</b>
+ - Once it is carried out, we can find the shortest path to all permanently labelled node.
+ - Only one diagram is enough to reflect all distances/paths.
+ - It is efficient enough to use for relatively large problems.
+ 
+<b>DISADVANTAGES</b>
+- It cannot handle negative weights which leads to acyclic graphs and most often cannot obtain the right shortest path.
+- It is a greedy algorithm that means it is possible for the algorithm to select the current best option which can make the algorithm get sidetracked following a potential path that doesn’t exist, simply because the edges along it form a short path.
+
+
+## Reference
+
+Shivani Sanan, Leena jain, Bharti Kappor (2013). (IJAIEM) "Shortest Path Algorithm" <br>
+https://www.ijaiem.org/volume2issue7/IJAIEM-2013-07-23-079.pdf
```

Comment:
Why is this paper cited in references? This doesn't define the Dijkstra algorithm.
It's defined in:
```
Dijkstra, Edsger W. "A note on two problems in connexion with graphs." Numerische mathematik 1, no. 1 (1959): 269-271.
```
https://ir.cwi.nl/pub/9256/9256D.pdf